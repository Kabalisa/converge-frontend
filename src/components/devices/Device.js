import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const formatDate = date => moment(date).format('DD MMM YYYY');

/**
 *
 * Device component
 *
 * @param {Object} device
 *
 * @returns {JSX}
 */
const Device = ({
  device: {
    name, deviceType, dateAdded, lastSeen, location, /* room: { name: roomName }, */
  },
}) => (
  <div className="table__row">
    <span>{name}</span>
    <span>{deviceType}</span>
    <span>{formatDate(dateAdded)}</span>
    <span>{formatDate(lastSeen)}</span>
    <span>{location}</span>
    {/* <span>{roomName}</span> */}
  </div>
);

Device.propTypes = {
  device: PropTypes.shape({
    name: PropTypes.string.isRequired,
    deviceType: PropTypes.string.isRequired,
    dateAdded: PropTypes.string.isRequired,
    lastSeen: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
  }).isRequired,
};

export default Device;
