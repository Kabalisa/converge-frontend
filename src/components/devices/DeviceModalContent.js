import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Input, SelectInput, ActionButtons } from '../commons';

const DeviceModalContent = ({
  formData: {
    name, deviceType, roomId,
  },
  methods: {
    handleNameChange, handleTypeChange, handleRoomChange, closeModal, submit,
  },
  modalContent, rooms, deviceTypes, fetching,
}) => (
  <form className="amenity-form device-type">
    <div>
      {
        modalContent !== 'delete'
          ? (
            <Fragment>
              <Input
                id="device-name"
                name="device-name"
                placeholder="Enter Device Name"
                labelName="Device Name"
                labelClass="device-input"
                value={name}
                onChange={handleNameChange}
              />
              <SelectInput
                labelText="Device Type"
                name="device-type"
                id="add-devices-type"
                isValue
                value={deviceType || 'Select Device'}
                options={deviceTypes}
                onChange={handleTypeChange}
                wrapperClassName="device-input"
                placeholder="Select Device"
                selectInputClassName="dynamic-input-field default-select"
              />
              <SelectInput
                labelText="Room"
                name="room"
                id="add-devices-rooms"
                value={roomId || 'Select Room'}
                options={rooms}
                onChange={handleRoomChange}
                wrapperClassName="device-input"
                placeholder="Select Room"
                selectInputClassName="dynamic-input-field default-select"
              />
            </Fragment>
          ) : (
            <p id="confirm-msg">
              Are you sure you want to delete &quot;{`${name}: ${deviceType}`}
              &quot;? <br />
              This cannot be undone
            </p>
          )
      }
      <div className="loading-btn-div">
        <ActionButtons
          withCancel
          isLoading={fetching}
          onClickCancel={closeModal}
          actionButtonText={modalContent && ((modalContent === 'edit' ? 'SAVE CHANGES' : `${modalContent.toUpperCase()} DEVICE`) || 'ADD DEVICE')}
          onClickSubmit={submit}
        />
      </div>
    </div>
  </form>
);

DeviceModalContent.propTypes = {
  formData: PropTypes.shape({
    name: PropTypes.string,
    roomId: PropTypes.string,
    deviceType: PropTypes.string,
  }).isRequired,
  methods: PropTypes.shape({
    handleNameChange: PropTypes.func,
    handleRoomChange: PropTypes.func,
    handleTypeChange: PropTypes.func,
    closeModal: PropTypes.func,
    submit: PropTypes.func,
  }).isRequired,
  fetching: PropTypes.bool,
  modalContent: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  rooms: PropTypes.instanceOf(Array).isRequired,
  deviceTypes: PropTypes.instanceOf(Array).isRequired,
};

DeviceModalContent.defaultProps = {
  fetching: false,
  modalContent: undefined,
};

export default DeviceModalContent;
