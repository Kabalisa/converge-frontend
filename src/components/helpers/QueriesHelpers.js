import { GET_USER_QUERY } from '../../graphql/queries/People';
import {
  GET_ALL_ROOMS,
  GET_LOCATIONS_QUERY,
  GET_ROOMS_QUERY,
  GET_ALL_REMOTE_ROOMS,
  GET_REMOTE_ROOMS_ALL_LOCATIONS,
} from '../../graphql/queries/Rooms';
import { GET_RESOURCES_QUERY } from '../../graphql/queries/Resources';
import ALL_ANALYTICS, { ANALYTICS_FOR_DAILY_ROOM_EVENTS } from '../../graphql/queries/analytics';
import { decodeTokenAndGetUserData } from '../../utils/Cookie';
import apolloClient from '../../utils/ApolloClient';
import GET_ALL_LEVELS from '../../graphql/queries/Levels';
import GET_ROOM_FEEDBACK_QUESTIONS_QUERY from '../../graphql/queries/questions';
import { GET_DEVICES_QUERY } from '../../graphql/queries/Devices';

const getUserDetails = async (client = apolloClient) => {
  const { UserInfo: userData } = decodeTokenAndGetUserData() || {};
  const email = process.env.NODE_ENV === 'test' ? 'converge@andela.com' : userData.email;
  try {
    const data = client.readQuery(
      {
        query: GET_USER_QUERY,
        variables: {
          email,
        },
      },
      true,
    );
    const user = Object.assign({}, data.user);
    user.firstName = userData.firstName;
    return user;
  } catch (err) {
    const { data } = await client.query({
      query: GET_USER_QUERY,
      variables: { email },
    });
    const user = Object.assign({}, data.user);
    user.firstName = userData.firstName;
    return user;
  }
};

const getAllLocations = async (client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: GET_LOCATIONS_QUERY,
      },
      true,
    );
    return data.allLocations;
  } catch (err) {
    const { data } = await client.query({
      query: GET_LOCATIONS_QUERY,
    });
    return data.allLocations;
  }
};

const getRoomList = async (userLocation, perPage, page, client = apolloClient) => {
  const { data } = await client.query({
    query: GET_ROOMS_QUERY,
    variables: {
      location: userLocation,
      office: '',
      page,
      perPage,
    },
  });
  return data;
};

const getAllResources = async (perPage, page, client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: GET_RESOURCES_QUERY,
        variables: {
          page: page || 1,
          perPage: perPage || 5,
        },
      },
      true,
    );
    return data.allResources;
  } catch (error) {
    const { data } = await client.query({
      query: GET_RESOURCES_QUERY,
      variables: {
        page: page || 1,
        perPage: perPage || 5,
      },
    });
    return data.allResources;
  }
};

const getAllRemoteRooms = async (client = apolloClient) => {
  const { data } = await client.query(
    {
      query: GET_ALL_REMOTE_ROOMS,
    },
    true,
  );
  return data.allRemoteRooms;
};

const getRemoteRoomsAllLocations = async (client = apolloClient) => {
  const { data } = await client.query(
    {
      query: GET_REMOTE_ROOMS_ALL_LOCATIONS,
      variables: {
        returnAll: true,
      },
    },
    true,
  );
  return data.allRemoteRooms;
};

const getRoomsStructure = async (client = apolloClient) => {
  const { data } = await client.query(
    {
      query: GET_ALL_LEVELS,
    },
    true,
  );
  return data;
};
const getRoomFeedbackQuestions = async (client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: GET_ROOM_FEEDBACK_QUESTIONS_QUERY,
      },
      true,
    );
    return data.questions;
  } catch (err) {
    const { data } = await client.query({
      query: GET_ROOM_FEEDBACK_QUESTIONS_QUERY,
    });
    return data.questions;
  }
};

const getAllRooms = async (location, client = apolloClient) => {
  try {
    const data = client.readQuery({
      query: GET_ALL_ROOMS,
      variables: {
        location,
      },
    }, true);
    return data;
  } catch (error) {
    const { data } = await client.query({
      query: GET_ALL_ROOMS,
      variables: {
        location,
      },
    }, true);
    return data;
  }
};

const getAllAnalytics = async (dateValue, client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: ALL_ANALYTICS,
        variables: dateValue,
      },
      true,
    );
    return data;
  } catch (err) {
    const { data } = await client.query({
      query: ALL_ANALYTICS,
      variables: dateValue,
    });
    return data;
  }
};

const query = (dateValue) => {
  const { startDate, endDate } = dateValue;
  return ({
    query: ANALYTICS_FOR_DAILY_ROOM_EVENTS,
    variables: {
      startDate,
      endDate,
    },
  });
};

const getAnalyticForDailyRoomsEvents
 = async (dateValue, client = apolloClient) => {
   try {
     const data = client.readQuery(
       query(dateValue),
       true,
     );
     return data;
   } catch (err) {
     const { data } = await client.query(
       query(dateValue),
     );
     return data;
   }
 };

const getAllDevices = async (client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: GET_DEVICES_QUERY,
      },
      true,
    );
    return data.allDevices;
  } catch (error) {
    const { data } = await client.query({
      query: GET_DEVICES_QUERY,
    });
    return data.allDevices;
  }
};

export {
  getAnalyticForDailyRoomsEvents,
  getAllAnalytics,
  getAllRooms,
  getUserDetails,
  getAllLocations,
  getAllResources,
  getAllRemoteRooms,
  getRoomList,
  getRoomsStructure,
  getRoomFeedbackQuestions,
  getRemoteRoomsAllLocations,
  getAllDevices,
};
