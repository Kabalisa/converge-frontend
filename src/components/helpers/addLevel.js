import Joi from 'joi-browser';

const AddLevel = (flattenedData, levelType) => {
  const Data = flattenedData.map((item) => {
    const { name } = item;
    return { name };
  });
  if (!levelType) {
    return 'Leveltype can not be empty';
  }

  const schema = {
    name: Joi.string().trim().min(2).required()
      .label(levelType),
  };
  const receivedErrors = Data.map((item) => {
    const { error: joiError } = Joi.validate(item, schema);
    if (joiError) {
      return joiError.details[0].message;
    }
    return '';
  });
  const receivedError = receivedErrors.find(item => item !== '');

  if (receivedError) {
    return receivedError;
  }
  return '';
};

export default AddLevel;
