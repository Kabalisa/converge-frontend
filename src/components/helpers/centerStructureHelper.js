/* eslint-disable no-await-in-loop */
/* eslint-disable no-restricted-syntax */
import {
  getRoomsStructure,
  getRoomId,
  getRoomResources,
} from './QueriesHelpers';

const isResourcesSetup = async (structureRooms) => {
  const roomIds = [];
  for (const room of structureRooms) {
    const roomId = await getRoomId(room.name);
    roomIds.push(roomId.data.getRoomByName[0]);
  }

  let allRoomsResources = [];

  for (const oneRoom of roomIds) {
    try {
      const oneRoomResource = await getRoomResources(oneRoom.id);
      allRoomsResources.push(oneRoomResource);
    } catch (error) {
      allRoomsResources = [[]];
    }
  }
  const levelResources = allRoomsResources
    .map(roomResources => roomResources.length > 0)
    .includes(true);
  return levelResources;
};
const Levels = (structure, level) =>
  structure.map(element => element.level === level).includes(true);
const LevelFilter = (structure, level) =>
  structure.filter(element => element.level === level);
const centerSetupLevel = async () => {
  const centerStructure = await getRoomsStructure();
  const levelOne = Levels(centerStructure.allStructures, 1);
  const levelFloors = Levels(centerStructure.allStructures, 2);
  const levelWings = Levels(centerStructure.allStructures, 3);
  const levelRooms = Levels(centerStructure.allStructures, 4);
  const structureRooms = LevelFilter(centerStructure.allStructures, 4);
  let levelResources;
  if (structureRooms.length) {
    levelResources = await isResourcesSetup(structureRooms);
  }
  // console.log('===========>levelResources', levelResources);
  if (!levelOne) {
    return 'welcomePage';
  }
  if (!levelFloors && !levelWings && !levelRooms && !levelResources) {
    return 'officeStructure';
  }
  if (!levelRooms && !levelResources) {
    return 'addRooms';
  }
  if (!levelResources) {
    return 'addResources';
  }
  return 'centerIsSetup';
};
export default centerSetupLevel;
