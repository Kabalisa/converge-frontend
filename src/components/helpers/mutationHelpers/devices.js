/* eslint-disable import/prefer-default-export */
import apolloClient from '../../../utils/ApolloClient';
import { ADD_DEVICE_MUTATION } from '../../../graphql/mutations/Devices';
import { GET_DEVICES_QUERY } from '../../../graphql/queries/Devices';

export const addDeviceMutation = async ({
  name, roomId, deviceType, location,
}, client = apolloClient) => {
  await client
    .mutate({
      mutation: ADD_DEVICE_MUTATION,
      name: 'addDeviceMutation',
      variables: {
        name, deviceType, roomId, location,
      },
      // write and read from cache
      update: async (proxy, { data: { createDevice } }) => {
        const cachedDevices = proxy.readQuery({
          query: GET_DEVICES_QUERY,
        });
        const deviceList = cachedDevices.allDevices;
        cachedDevices.allDevices = [...deviceList, createDevice.device];

        proxy.writeQuery({
          query: GET_DEVICES_QUERY,
          data: cachedDevices,
        });
      },
    });
};
