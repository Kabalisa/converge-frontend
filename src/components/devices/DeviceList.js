import React from 'react';
import '../../assets/styles/devicelist.scss';
import ColGroup from '../helpers/ColGroup';
import TableHead from '../helpers/TableHead';
import Device from './Device';
import devices from '../../fixtures/devices';
import MenuTitle from '../commons/MenuTitle';

const deviceListItems = devices.map(device => (
  <Device device={device} key={device.name} />
));

/**
 * Device List Component
 *
 * @returns {JSX}
 */
const DeviceList = () => (
  <div className="settings-devices-list">
    <div className="settings-devices-control">
      <MenuTitle title="Devices" />
    </div>
    <table>
      <ColGroup />
      <TableHead
        titles={['Name', 'Type', 'Date Added', 'Last Seen', 'Location']}
      />
      <tbody>{deviceListItems}</tbody>
    </table>
  </div>
);

export default DeviceList;
