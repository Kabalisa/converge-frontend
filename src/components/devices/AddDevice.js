import React, { Component } from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import notification from '../../utils/notification';
import MrmModal from '../commons/Modal';
import { ActionButtons, Input, SelectInput } from '../commons';
import { addDeviceMutation } from '../helpers/mutationHelpers/devices';

class DeviceFormModal extends Component {
  state = {
    isModalClosed: false,
    name: '',
    roomId: '',
    deviceType: '',
  }

  componentDidMount() {
    const { device } = this.props;
    if (device) this.updateStateWithPropDevice();
  }

  getDeviceTypes = () => ([{
    name: '7"',
  }, {
    name: '8"',
  }, {
    name: '9.7"',
  }, {
    name: '10.1"',
  }])

  updateStateWithPropDevice = () => {
    const { name, roomId, deviceType } = this.props.device;
    this.setState({
      name, roomId, deviceType,
    });
  }

  handleNameChange = ({ target: { value: name } }) => {
    this.setState({ name });
  };

  handleTypeChange = ({ target: { value: deviceType } }) => {
    this.setState({ deviceType });
  };

  handleRoomChange = ({ target: { value: roomId } }) => {
    this.setState({ roomId });
  };

  closeModal = (e) => {
    if (e) {
      e.preventDefault();
    }
    this.setState({ isModalClosed: true });
  };

  handleModalCloseRequest = () => {
    this.state.isModalClosed && this.setState({ isModalClosed: false });
  };

  submit = async (e) => {
    e.preventDefault();
    const { name, roomId, deviceType } = this.state;
    const { location, refetch } = this.props;
    if (
      !name || !roomId || !deviceType
    ) {
      return notification(toastr, 'error', 'Please fill all fields')();
    }
    try {
      this.setState({ fetching: true });
      await addDeviceMutation({
        name, roomId, location: location.name, deviceType,
      });
      await refetch();
      this.setState({
        fetching: false,
        isModalClosed: true,
        name: '',
        deviceType: '',
      });

      return notification(toastr, 'success', 'Device successfully added.')();
    } catch (error) {
      notification(toastr, 'error', 'An error occured while adding device.')();
      return this.setState({ fetching: false });
    }
  };

  render() {
    const {
      isModalClosed, name, fetching,
    } = this.state;
    const { rooms } = this.props;
    return (
      <MrmModal
        title="ADD A NEW DEVICE"
        buttonText="Add a New Device"
        closeModal={isModalClosed}
        handleCloseRequest={this.handleModalCloseRequest}
        className="add-resource-modal"
      >
        <form className="amenity-form device-type">
          <div>
            <Input
              id="device-name"
              name="device-name"
              placeholder="Enter Device Name"
              labelName="Device Name"
              labelClass="device-input"
              value={name}
              onChange={this.handleNameChange}
            />
            <SelectInput
              labelText="Device Type"
              name="device-type"
              id="add-devices-type"
              isValue
              value="Select Device"
              options={this.getDeviceTypes()}
              onChange={this.handleTypeChange}
              wrapperClassName="device-input"
              placeholder="Device Type"
              selectInputClassName="dynamic-input-field default-select"
            />
            <SelectInput
              labelText="Room"
              name="room"
              id="add-devices-rooms"
              value="Select Room"
              options={rooms}
              onChange={this.handleRoomChange}
              wrapperClassName="device-input"
              placeholder="Select Room"
              selectInputClassName="dynamic-input-field default-select"
            />
            <div className="loading-btn-div">
              <ActionButtons
                withCancel
                isLoading={fetching}
                onClickCancel={this.closeModal}
                actionButtonText="ADD DEVICE"
                onClickSubmit={this.submit}
              />
            </div>
          </div>
        </form>
      </MrmModal>
    );
  }
}

DeviceFormModal.propTypes = {
  refetch: PropTypes.func.isRequired,
  location: PropTypes.shape({
    name: PropTypes.string,
  }),
  rooms: PropTypes.instanceOf(Array),
  device: PropTypes.shape({
    name: PropTypes.string,
    deviceType: PropTypes.string,
    roomId: PropTypes.number,
  }),
};

DeviceFormModal.defaultProps = {
  location: undefined,
  rooms: [],
  device: undefined,
};

export default DeviceFormModal;
