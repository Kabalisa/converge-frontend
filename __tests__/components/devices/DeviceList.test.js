import React from 'react';
import { shallow } from 'enzyme';
import DeviceList from '../../../src/components/devices/DeviceList';
import { getAllDevices, getAllRooms } from '../../../src/components/helpers/QueriesHelpers';
import devices from '../../../src/fixtures/devices';

const moreDevices = [{
  name: 'Samsung tablet',
  deviceType: 'Internal Display',
  dateAdded: '14 Aug 2018',
  lastSeen: '16 Feb 2018',
  location: 'Nairobi',
}, {
  name: 'Samsung Galaxy',
  deviceType: 'Internal Display',
  dateAdded: '14 Aug 2018',
  lastSeen: '16 Feb 2018',
  location: 'Nairobi',

}];


jest.mock('../../../src/components/helpers/QueriesHelpers');

describe('DeviceList Component', () => {
  let wrapper;
  beforeAll(() => {
    getAllDevices.mockResolvedValue(devices);
    getAllRooms.mockResolvedValue({
      allRooms: {
        rooms: [],
      },
    });
    wrapper = shallow(<DeviceList location={{
      name: 'Lagos',
    }}
    />);
  });
  it('renders correctly from memory', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('updatePageDetails behaves as expected', () => {
    wrapper.instance().updatePageDetails();
    expect(wrapper.state('hasNext')).toBe(false);
  });

  it('handles page change correctly', () => {
    const mock = jest.spyOn(wrapper.instance(), 'updatePageDetails');
    wrapper.instance().handlePageChange(4, 1);
    wrapper.instance().handlePageChange(undefined, 1);
    expect(wrapper.state('currentPage')).toBe(1);
    expect(mock).toHaveBeenCalled();
  });

  it('getData method works as expected', async () => {
    await wrapper.instance().getData();
    expect(getAllDevices).toHaveBeenCalled();
    expect(getAllRooms).toHaveBeenCalled();
  });

  it('renders error icon', async () => {
    getAllDevices.mockResolvedValue([]);
    wrapper = shallow(<DeviceList location={{
      name: 'Lagos',
    }}
    />);
    await wrapper.instance().getData();
    expect(wrapper.exists('ErrorIcon')).toBeTruthy();
  });

  it('renders pagination component', async () => {
    getAllDevices.mockResolvedValue(devices.concat(moreDevices));
    wrapper = shallow(<DeviceList location={{
      name: 'Lagos',
    }}
    />);
    await wrapper.instance().getData();
    expect(wrapper.exists('Pagination')).toBeTruthy();
  });
});
