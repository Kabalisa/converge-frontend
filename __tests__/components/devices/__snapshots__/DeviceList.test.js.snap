// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceList Component renders correctly from memory 1`] = `
<div
  className="settings-devices-list overlay-container"
>
  <div
    className="settings-devices-control"
  />
  <div
    className="add-new-resource"
  >
    <DeviceFormModal
      location={
        Object {
          "name": "Lagos",
        }
      }
      refetch={[Function]}
      rooms={Array []}
    />
  </div>
  <div
    className="table"
  >
    <TableHead
      titles={
        Array [
          "Name",
          "Type",
          "Date Added",
          "Last Seen",
          "Location",
        ]
      }
    />
    <div
      className="table__body"
    >
      <Device
        device={
          Object {
            "dateAdded": "14 Aug 2018",
            "deviceType": "Internal Display",
            "lastSeen": "16 Feb 2018",
            "location": "Nairobi",
            "name": "Samsung Note",
          }
        }
        key="Samsung Noteundefined"
      />
      <Device
        device={
          Object {
            "dateAdded": "14 Aug 2018",
            "deviceType": "Internal Display",
            "lastSeen": "16 Feb 2018",
            "location": "Nairobi",
            "name": "Samsung duos",
          }
        }
        key="Samsung duosundefined"
      />
      <Device
        device={
          Object {
            "dateAdded": "14 Aug 2018",
            "deviceType": "Internal Display",
            "lastSeen": "16 Feb 2018",
            "location": "Nairobi",
            "name": "Samsung Galaxy",
          }
        }
        key="Samsung Galaxyundefined"
      />
      <Device
        device={
          Object {
            "dateAdded": "14 Aug 2018",
            "deviceType": "Internal Display",
            "lastSeen": "16 Feb 2018",
            "location": "Nairobi",
            "name": "Samsung tablet",
          }
        }
        key="Samsung tabletundefined"
      />
    </div>
  </div>
</div>
`;
